### PR TITLE
WOD-26 Implement WebView UI

### PR DESCRIPTION
**Pull request by mickeylee25**
https://livestyled.atlassian.net/browse/WOD-26
This PR only creates the UI, the styling and interfaces will be in the following PRs.

**New**
* OrderingJourneyView: A SwiftUI view contains WebViewWrapper, navigation bar, and bottom views.
* OrderingJourneyViewController: A UIHostController to bridge UIKit. This class is created for dismissing WOJ from the UIKit apps because SwiftUI view cannot be dismissed from UIKit due to unsupported `presentationMode` until iOS 15.
* WebViewWrapper: A wrapper to transform WKWebView to SwiftUI, including WKNavigationDelegate methods.
* WebViewStore: A view model layer object to the wrapper.
* Add ViewInspector pod for unit testing SwiftUI view. [GitHub repo](https://github.com/nalexn/ViewInspector)

**TODO**
1. Improve coverage on WebViewWrapper
1.1 I was struggling to mock Context of `func makeUIView(context: Context) -> WKWebView` as it's returned from UIKit.
1.2 `parent.store.webViewNavigationPublisher.sink(receiveValue:` doesn't get called when the subject is mocked. Need to investigate it.
2. Improve coverage on OrderingJourneyView
2.1 `.onReceive(store.canGoBack.receive(on:))`
2.2 `.onReceive(store.canGoForward.receive(on:))`
2.3 Buttons' actions.

Screenshot 2021-08-19 at 14 41 06

### CODE REVIEW

File: Analytics/AnalyticsLogging/AnalyticsLogger.swift [hunk 1]
```diff
@@ -7,7 +7,9 @@
 //
 
 import Foundation
+#if !COCOAPODS
```
Comment (mickeylee25) [inline Analytics/AnalyticsLogging/AnalyticsLogger.swift:10]: This change is caused by cherry-pick the import GraphQL error from another branch, please ignore in this PR.

File: DummyProject/DummyProject/ViewController.swift [hunk 2]
```diff
@@ -66,16 +67,18 @@ class ViewController: UIViewController {
...
-            switch result {
-            case .success(let isMember):
-                showAlertController(title: "Audience", message: "Success with result isMember: \(isMember)")
-            case .failure(let error):
-                showAlertController(title: "Audience", message: "Error with: \(error.localizedDescription)")
-            }
-        }
+        let controller = RealifeTech.Sell.createOrderingJourneyViewController()
+        present(controller, animated: true, completion: nil)
+//        checkIfSDKIsReady()
```
Comment (mickeylee25) [inline DummyProject/DummyProject/ViewController.swift:72]: This will be resolved in WOD-31

File: Sell/OrderingJourney/OrderingJourneyView.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import WebKit
+
+public struct OrderingJourneyView: View {
+
+    @ObservedObject private var store = WebViewStore()
+    @State private var canGoBack = false
+    @State private var canGoForward = false
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+
+    var dismiss: (() -> Void)?
```
Comment (mickeylee25) [inline Sell/OrderingJourney/OrderingJourneyView.swift:19]: [Note] This is only assigned from UIHostController as the presentationMode doesn't support dismissing from UIKit until iOS 15.
https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-15-beta-release-notes#Updates-in-iOS-&-iPadOS-15-beta
> isPresented, PresentationMode, and the new DismissAction action dismiss a hosting controller presented from UIKit. (52556186)

File: Sell/OrderingJourney/OrderingJourneyView.swift [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    public let urlRequest: URLRequest
+
+    public init(urlString: String) {
+        guard let url = URL(string: urlString) else {
```
Comment (mickeylee25) [inline Sell/OrderingJourney/OrderingJourneyView.swift:24]: The initialiser will be tweaked in WOD-28

File: Core/GraphQL/Tests/RequestLoggingInterceptorTests.swift [hunk 5]
```diff
@@ -43,7 +43,7 @@ final class RequestLoggingInterceptorTests: XCTestCase {
         ) { _ in
             expectation.fulfill()
         }
-        wait(for: [expectation], timeout: 0.01)
+        wait(for: [expectation], timeout: 0.5)
```
Comment (You-Hsuan) [inline Core/GraphQL/Tests/RequestLoggingInterceptorTests.swift:46]: May I know the reason for increasing timeout please?

File: Sell/OrderingJourney/Tests/WebViewWrapperTests.swift [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+//    func test_coordinatorDidStartProvisionalNavigation() {
+//        let wkNavigation = createNavigation(url: url, webView: webView)
+//        let coordinator = sut.makeCoordinator()
+//        coordinator.webView(
+//            webView,
+//            didStartProvisionalNavigation: wkNavigation)
+//        store.webViewNavigationPublisher.send(.reload)
+//        XCTAssertTrue(webView.didReload)
+//    }
```
Comment (You-Hsuan) [inline Sell/OrderingJourney/Tests/WebViewWrapperTests.swift:74]: Could this be removed?

File: Core/GraphQL/Tests/RequestLoggingInterceptorTests.swift (hunk 5 above)
Comment (mickeylee25) [reply Core/GraphQL/Tests/RequestLoggingInterceptorTests.swift:46]: Flaky test

**Review by You-Hsuan — APPROVED** [review]

**Pull request merged**